### PR TITLE
Use new identifier for default objects

### PR DESCRIPTION
**Pull request by Gymnasiast**

### CODE REVIEW

File: src/openrct2/EditorObjectSelectionSession.h [hunk 1]
```diff
@@ -16,6 +16,7 @@
 
 enum EDITOR_INPUT_FLAGS
 {
+    INPUT_FLAG_EDITOR_OBJECT_0 = (1 << 0),
```
Comment (Gymnasiast) [inline src/openrct2/EditorObjectSelectionSession.h:19]: This flag is used to select an object - leaving it out deselects it.

File: src/openrct2/EditorObjectSelectionSession.h [hunk 2]
```diff
@@ -16,6 +16,7 @@
...
     INPUT_FLAG_EDITOR_OBJECT_1 = (1 << 1),
     INPUT_FLAG_EDITOR_OBJECT_2 = (1 << 2),
```
Comment (Gymnasiast) [inline src/openrct2/EditorObjectSelectionSession.h:21]: This flag is used to select all objects from a scenario group.

File: src/openrct2/EditorObjectSelectionSession.cpp [hunk 3]
```diff
@@ -342,9 +342,13 @@ static void window_editor_object_selection_select_default_objects()
 {
     if (_numSelectedObjectsForType[0] == 0)
     {
-        for (const auto& defaultSelectedObject : DefaultSelectedObjects)
+        for (auto defaultSelectedObject : DefaultSelectedObjects)
```
Comment (duncanspumpkin) [inline src/openrct2/EditorObjectSelectionSession.cpp:345]: This should still be `const`

**Review by IntelOrca — APPROVED** [review]

File: src/openrct2/EditorObjectSelectionSession.cpp (hunk 3 above)
Comment (tupaschoal) [reply src/openrct2/EditorObjectSelectionSession.cpp:345]: This was marked as resolved but still isn't `const` as before

File: src/openrct2/EditorObjectSelectionSession.cpp [hunk 4]
```diff
@@ -408,16 +412,15 @@ static void set_object_selection_error(uint8_t is_master_object, rct_string_id e
  *
  *  rct2: 0x006AB54F
  */
-int32_t window_editor_object_selection_select_object(uint8_t bh, int32_t flags, const rct_object_entry* entry)
+bool window_editor_object_selection_select_object(uint8_t isMasterObject, int32_t flags, const ObjectRepositoryItem* item)
```
Comment (tupaschoal) [inline src/openrct2/EditorObjectSelectionSession.cpp:415]: What is a "master object"? Seems weird to have something that "isSomething" but isn't a boolean.

File: src/openrct2/EditorObjectSelectionSession.cpp [hunk 5]
```diff
@@ -483,44 +486,57 @@ int32_t window_editor_object_selection_select_object(uint8_t bh, int32_t flags,
...
 
         _numSelectedObjectsForType[objectType]++;
 
         *selectionFlags |= OBJECT_SELECTION_FLAG_SELECTED;
-        return 1;
+        return true;
     }
 }
 
+bool window_editor_object_selection_select_object(uint8_t isMasterObject, int32_t flags, std::string_view identifier)
```
Comment (tupaschoal) [inline src/openrct2/EditorObjectSelectionSession.cpp:527]: Should this be title case?

File: src/openrct2/EditorObjectSelectionSession.cpp [hunk 6]
```diff
@@ -483,44 +486,57 @@ int32_t window_editor_object_selection_select_object(uint8_t bh, int32_t flags,
...
+{
+    auto& objectRepository = OpenRCT2::GetContext()->GetObjectRepository();
+    const auto* item = objectRepository.FindObject(identifier);
+    return window_editor_object_selection_select_object(isMasterObject, flags, item);
+}
+
+bool window_editor_object_selection_select_object(uint8_t isMasterObject, int32_t flags, const rct_object_entry* entry)
```
Comment (tupaschoal) [inline src/openrct2/EditorObjectSelectionSession.cpp:534]: Should this be title case?

File: src/openrct2/EditorObjectSelectionSession.h [hunk 7]
```diff
@@ -16,8 +16,9 @@
 
 enum EDITOR_INPUT_FLAGS
 {
+    INPUT_FLAG_EDITOR_OBJECT_SELECT = (1 << 0), // Set when you want to select an object, not set when you want to deselect it.
```
Comment (tupaschoal) [inline src/openrct2/EditorObjectSelectionSession.h:19]:
```suggestion
    INPUT_FLAG_EDITOR_OBJECT_SELECT = (1 << 0), // Set when you want to select an object, unset when you want to deselect it.
```

File: src/openrct2/EditorObjectSelectionSession.cpp (hunk 4 above)
Comment (Gymnasiast) [reply src/openrct2/EditorObjectSelectionSession.cpp:415]: I took the name from the underlying function. I'm not 100% sure it is correct, which is why I didn't make any changes to the data type yet.

File: src/openrct2/EditorObjectSelectionSession.cpp (hunk 5 above)
Comment (Gymnasiast) [reply src/openrct2/EditorObjectSelectionSession.cpp:527]: Perhaps better taken as part of a window cleanup. (Would especially be nice to make it a method if we were to use subclassing.)

File: src/openrct2/EditorObjectSelectionSession.cpp (hunk 4 above)
Comment (tupaschoal) [reply src/openrct2/EditorObjectSelectionSession.cpp:415]: Ah, sure, I'm not questioning, I'm trying to understand because I have no clue hahaha

**Pull request merged**